src/taipy/core/_repository/_sql_repository.py

```diff
@@ -213,9 +252,238 @@ def _export(self, entity_id: str, folder_path: Union[str, pathlib.Path]):
 
         export_path = export_dir / f"{entity_id}.json"
 
-        entry = self.session.query(_TaipyModel).filter_by(model_id=entity_id).first()
+        entry = self.session.query(self._table).filter_by(model_id=entity_id).first()
         if entry is None:
             raise ModelNotFound(self.model, entity_id)  # type: ignore
 
         with open(export_path, "w", encoding="utf-8") as export_file:
             export_file.write(entry.document)
+
+
+class _TaipyVersionTable(_BaseSQLRepository):
+    @abstractmethod
+    def _to_model(self, obj):
+        """
+        Converts the object to be saved to its model.
+        """
+        ...
+
+    @abstractmethod
+    def _from_model(self, model):
+        """
+        Converts a model to its functional object.
+        """
+        ...
+
+    def __init__(
+        self,
+        model: Type[ModelType],
+        to_model_fct: Callable,
+        from_model_fct: Callable,
+    ):
+
+        self.model = model
+        self._to_model = to_model_fct  # type: ignore
+        self._from_model = from_model_fct  # type: ignore
+        self._table = _TaipyVersion
+        super().__init__()
+
+    def load(self, id: str) -> Entity:
+        entry = self.__get_by_id(id)
+        if entry is None:
+            raise ModelNotFound(self.model, id)  # type: ignore
+        return self.__to_entity(entry)
+
+    def _load_all(self, version_number: Optional[str] = None) -> List[Entity]:
+        try:
+            entries = self.session.query(self._table).all()
+            return [self.__to_entity(e) for e in entries]
+        except NoResultFound:
+            return []
+
+    def _load_all_by(self, by, version_number: Optional[str] = None) -> List[Entity]:
```

Comment:
Same here, I think we are not using the `version_number`.

Reply:
I found the following are not using version_number: 
- `_load_all`
- `_load_all_by`
- `_search`

I guess these will be tested on new tests on the other PR.

Reply:
Nevermind, Giang already spotted them.
:/

Reply:
Same as I've said before. 

Yes, but it is a problem of the current repository design. Since we are forced to follow the contract for the repositories and in the AbstractRepository we have the _load_all to take the version number, so every signature needs to have it.